ROX-12336: Harden SCCs for sensor to prevent running as root or privilege escalation

vikin91 opened the pull request:
## Description

This PR hardens the security for sensor.
Based on the sensor-SCCs (that are optional in the installation), sensor was allowed to escalate the privileges from user 4000 (default) to 0 (root).
By changing this, we would most probably pass the validation that runs for `restricted-v2` SCC as defined [here](https://docs.openshift.com/container-platform/4.11/authentication/managing-security-context-constraints.html#default-sccs_configuring-internal-oauth).

The SCC being changed here was introduced [in this PR](https://github.com/stackrox/rox/pull/5129) most probably based on linter suggestions.
Creation of SCC resources was made optional in https://github.com/stackrox/rox/pull/8686.

## Checklist
- [x] Investigated and inspected CI test results
- [ ] ~Unit test and regression tests added~
- [x] Evaluated and added CHANGELOG entry if required
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

- [x] Manually changing SCC on OpenShift 4.11, recreating the pod and verifying that sensor still works
- [x] CI

vikin91 commented:
/retest

vikin91 commented:
/test all

mtesseract commented:
Hi. I am not too familiar with SCCs. What is the reason in the first place that we have a custom SCC instead of simply using e.g. the `restricted-v2` SCC?

Besides, if this hardening would pose any risk I would believe that this would be easy to catch in our test suite (because it would most likely render all applications for which this SCC is configured as dysfunctional, correct?). Do we have sufficient test coverage among different OpenShift versions for such a change?

porridge reviewed (commented):
This LGTM, except an "obscure side effect" of this change is that if any other customer's workload happened to be bound by this SCC (by mistake/conicidence), then this hardening could cause their workload to break.
Such things did happen in the past I think (cc @misberner ) so I think a note in the changelog is due.

vikin91 commented:
> Hi @vikin91. I am not too familiar with SCCs. What is the reason in the first place that we have a custom SCC instead of simply using e.g. the `restricted-v2` SCC?

I have tried to find the reason for custom SCCs and found only this what I mentioned in the PR description - most probably, we added them based on a linter suggestion and never looked at them again until [ROX-9342
](https://issues.redhat.com/browse/ROX-9342). In [ROX-9342
](https://issues.redhat.com/browse/ROX-9342), the reporter suggested that if we create a cluster with custom SCCs, then the SCCs suggest that sensor may run as root or may gain root privileges. In the practice this is not the case, as we protect ourselves with additional settings `readOnlyRootFilesystem: true` `runAsNonRoot: true` in the deployment definition. I thought that we could express this in the custom SCC.

I have introduced here the minimal changes to the previous state to achieve this, but I can also give it a try with simply using `restricted-v2` directly. However, there may be an issue related to that change, as `restricted-v2` was introduced in OpenShift 4.11, before we had only `restricted` available.

> Besides, if this hardening would pose any risk I would believe that this would be easy to catch in our test suite (because it would most likely render all applications for which this SCC is configured as dysfunctional, correct?). Do we have sufficient test coverage among different OpenShift versions for such a change?

I think, we do not have any tests for this, as the original issue was discovered by Sam Fowler who reported [ROX-9342
](https://issues.redhat.com/browse/ROX-9342) and not by our test suites. The change itself should work along all versions of openshift as long as we stay with custom SCC (`restricted-v2` was introduced only recently).

vikin91 commented:
> This LGTM, except an "obscure side effect" of this change is that if any other customer's workload happened to be bound by this SCC (by mistake/conicidence), then this hardening could cause their workload to break. Such things did happen in the past I think (cc @misberner ) so I think a note in the changelog is due.

The creation of these SCCs remains optional, to only a subset would be affected. But I agree about the changelog entry - will add one in a moment.

vikin91 commented:
Closing as the proper changes will be done in https://github.com/stackrox/stackrox/pull/2928 and its follow-ups.
